### DISCUSSION TITLE
AGI Is Pointless

### DISCUSSION THREAD

**Discussion by giscus** [Andrew-Chen-Wang/blog#5] [General]
# AGI Is Pointless

Explaining what we should be focusing on and worried about, if at all

https://blog.andrewcwang.com/blog/2024-10-26-agi-is-pointless/

**Comment by Andrew-Chen-Wang** [comment]
I had the opportunity to try out Devin.ai before their public launch. I didn't get to [say a lot of the disparaging things (it was all edited out :/)](https://www.lumos.com/blog/cognitions-devin-your-new-junior-engineer) to maintain a good relationship with the DevinAI team. The two teams I led didn't find much practical use in this enterprise product, and we did try to use the product intended by the Cognition product and engineering teams.

For one, it required a lot of business context. Interestingly, when we were first introduced, they pitched me on how versatile DevinAI is such as getting an airplane refund, but when we asked deeper what most customers use it for, they found "repetitive" processes the most useful. Have you seen the video of a man doing some hand movements in front of a robot, and then the robot mimics the movements in repetition? That's what DevinAI wanted us to do. So to get DevinAI ready, we had to set up everything ourselves. DevinAI got stuck on a lot of steps including our MacOS setup.sh script where DevinAI couldn't figure out how to do a linux version of the setup.sh script (i.e. install an alternative to our homebrew installations).

I tried to mention this, but unlike other AI agent products where they require you to start from scratch, I was impressed that DevinAI can take in a huge code base, generate a plan, and actually work. It seamlessly integrated with the CI to make a feedback loop as well. I highly valued the integrations and tooling that DevinAI had, particularly the VSCode integration where I could SSH into the Ubuntu VM and DevinAI could navigate the web on its own (and I could spectate!). The Replit AI UI is better and have similar features, but the thing that both of these tools don't do (by confirmation from the team) and another startup called DepsHub is reading your installed package's source code.

Part 2 of my disappointment with LLMs: they require deep integration with data. For DevinAI, they still need to manually figure out how to get DevinAI to visualize and contextualize the frontend to make sure that it's working as intended by the developer. In the case of coders, the integration comes from knowledge of what the code does and, well, knowledge of the code. DepsHub, ReplitAI, DevinAI, etc. don't actually read your installed packages, so they're essentially guessing how to write code and get function signatures wrong all the time.

Even if you make a customer service AI agent, you require a lot of integration and good structure from your CX playbook being neatly written to sourcing bug tickets in Jira to tell a customer when a bug will be resolved. The more integrations, the more AI agents begin to look like a B2B SaaS.

The bottom line is that getting an AI agent to work... takes a lot of work. The world is human and business focused (even in customer service, sometimes, even if you're told that a bug is fixed, you have to wait for the bug to deploy, but the Jira ticket doesn't have that context, only the developer. So a customer service rep would ask the developer, but an AI agent may not. That requires manual programming to tell an AI customer service rep to do that. That is a good startup idea though ha; a marketplace of scripts for your AI agent to use on how to deal with customers given your business context).

**Comment by Andrew-Chen-Wang** [comment]
One more thing is that LLMs specifically are very good at: specifically analysis and interpretation, how integrations are highly valuable for AI to complete complex tasks, human flexibility for modifying AI responses, and shaping UI design to empower clients in how they manage reputation concerns regarding how an AI customer service representative replies generatively and to what extent (e.g. speed, accuracy, handling rejection, threshold for human escalation).
